metadata/src/main/java/org/apache/kafka/metadata/placement/PlacementSpec.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.metadata.placement;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Objects;
+
+
+/**
+ * Specifies a replica placement that we want to make.
+ */
+@InterfaceStability.Unstable
```

Comment:
Should we include this hint on more things in `metadata` package? Is this intended as a signal to implementers of other components within Kafka (like in `core` package)?

Reply:
Technically, anything without an annotation is unstable. I just wanted to emphasize it here since there have been discussions about making placement policy pluggable in the past. (For the record I generally don't think it should be pluggable in the sense of 3rd party code implementing this interface)